crates/volta-core/src/tool/node/resolve.rs

```diff
@@ -249,15 +249,11 @@ fn resolve_node_versions(url: &str) -> Fallible<RawNodeIndex> {
             debug!("Node index cache was not found or was invalid");
             let spinner = progress_spinner(&format!("Fetching public registry: {}", url));
 
-            let (_, headers, response) = attohttpc::get(url)
-                .send()
-                .and_then(Response::error_for_status)
-                .with_context(registry_fetch_error("Node", url))?
-                .split();
+            let mut response: reqwest::blocking::Response =
+                reqwest::blocking::get(url).with_context(registry_fetch_error("Node", url))?;
 
-            let response_text = response
-                .text()
-                .with_context(registry_fetch_error("Node", url))?;
+            let mut response_text = String::new();
```

Comment:
I struggled with this code here. The goal is to make a blocking request and get both the response body and also the response headers which is needed lower in the file. Let me know if there's a better approach.

Reply:
Yeah, the lack of a way to destructure the `Response` makes this tricky. I think the easiest approach might be to determine the `expires` string _before_ reading the data with something like:
```rust
let expires = if let Ok(expires_header) = response.headers().decode::<Expires>() {
    expires_header.to_string()
} else {
    let expiry_date = SystemTime::now() + Duration::from_secs(max_age(&response).into());
    HttpDate::from(expiry_date).to_string()
};
```
Then we can use:
```
let response_text = response.text().with_context(registry_fetch_error("Node", url))?;
```
Here, and finally do `write!(expiry_file, "{}", expires).with_context(...)` at the end of the file. So we calculate everything we need from the response _first_, then consume it to get the raw text, and then use the pre-calculated bit later on when we need it.